hack/ensure_job_cluster.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+"""This script ensures the cluster of prowjobs is defined as expected.
+e.g., python ./hack/ensure_job_cluster.py
+"""
+
+import argparse
+import logging
+import os
+import yaml
+
+
+def get_desired_cluster(file_path, job):
+    """get the desired cluster for a given job defined in file_path."""
+    if job.get("agent", "") != "kubernetes":
+        return ""
+    if job["name"] in JOB_MAP:
+        return JOB_MAP[job["name"]]
+    if job["name"].endswith("-build01") or migrated(file_path):
+        return BUILD01_CLUSTER
+    return ""
+
+
+def get_updated_jobs(file_path, jobs, overwrite):
+    """get jobs with correct cluster."""
+    new_jobs = []
+    for job in jobs:
+        cluster = get_desired_cluster(file_path, job)
+        if cluster != job.get("cluster", ""):
+            message = "expected the cluster of job {} to be {}, but found {}"\
+                .format(job["name"], cluster, job.get("cluster", ""))
+            if not overwrite:
```

Comment:
So this approach still splits the responsibility -- you should be able to check whether or not there are updated jobs in the caller, and act on that as you will.